src/ray/gcs/gcs_server/gcs_server.cc

```diff
@@ -417,15 +423,42 @@ void GcsServer::StoreGcsServerAddressInRedis() {
 }
 
 void GcsServer::InitRaySyncer(const GcsInitData &gcs_init_data) {
-  /*
-    The current synchronization flow is:
-        raylet -> syncer::poller --> syncer::update -> gcs_resource_manager
-        gcs_placement_scheduler --/
-  */
-  ray_syncer_ = std::make_unique<gcs_syncer::RaySyncer>(
-      main_service_, raylet_client_pool_, *gcs_resource_manager_);
-  ray_syncer_->Initialize(gcs_init_data);
-  ray_syncer_->Start();
+  if (RayConfig::instance().use_ray_syncer()) {
+    gcs_node_id_ = NodeID::FromRandom();
+    ray_syncer_ = std::make_unique<syncer::RaySyncer>(ray_syncer_io_context_,
+                                                      gcs_node_id_.Binary());
+    ray_syncer_->Register(
+        syncer::RayComponentId::RESOURCE_MANAGER, nullptr, gcs_resource_manager_.get());
+    ray_syncer_->Register(
+        syncer::RayComponentId::SCHEDULER, nullptr, gcs_resource_manager_.get());
+    ray_syncer_thread_ = std::make_unique<std::thread>([this]() {
+      boost::asio::io_service::work work(ray_syncer_io_context_);
+      ray_syncer_io_context_.run();
+    });
+
+    for (const auto &pair : gcs_init_data.Nodes()) {
+      if (pair.second.state() ==
+          rpc::GcsNodeInfo_GcsNodeState::GcsNodeInfo_GcsNodeState_ALIVE) {
+        rpc::Address address;
+        address.set_raylet_id(pair.second.node_id());
+        address.set_ip_address(pair.second.node_manager_address());
+        address.set_port(pair.second.node_manager_port());
+
+        auto raylet_client = raylet_client_pool_->GetOrConnectByAddress(address);
+        ray_syncer_->Connect(raylet_client->GetChannel());
+      }
+    }
+  } else {
```

Comment:
What's the plan to remove the else case?

Reply:
- Tune the performance of ray syncer to make sure no regression
- Turn on the flag on by default
- Remove the old one. (targeting early June)